plugins/serializers/carbon2/carbon2_test.go

```diff
@@ -38,6 +38,11 @@ func TestSerializeMetricFloat(t *testing.T) {
 			format:   Carbon2FormatFieldSeparate,
 			expected: fmt.Sprintf("metric=cpu field=usage_idle cpu=cpu0  91.5 %d\n", now.Unix()),
 		},
+		{
```

Comment:
It doesn't make sense to me to duplicate the tests here.  Empty doesn't provide different functionality than Separate.  It would be simpler to change NewSerializer so that if metricsFormat argument is empty string then it returns a Serializer with metricsformat Carbon2FormatFieldSeparate.  Then you could remove Carbon2FormatFieldEmpty altogether.  Then you'd only need a single test that NewSerializer("") returns a Serializer with metricsFormat of Carbon2FormatFieldSeparate.

Reply:
Ok, sure.
So I've made `Carbon2FormatFieldEmpty` unexported (i.e. `carbon2FormatFieldEmpty`) and with this the only place where we need to check for empty string is the constructor as it's impossible to pass `format("")` even in `SetMetricsFormat()` because of `format` being unexported as well.